File: test/se/crisp/signup4/acceptance/pages/ErrorPage.java
package se.crisp.signup4.acceptance.pages;

import se.crisp.signup4.acceptance.PlayContainer;
import se.crisp.signup4.acceptance.SharedDriver;
import org.openqa.selenium.By;

public class ErrorPage {
  private final SharedDriver driver;

  public ErrorPage(SharedDriver driver) {
    this.driver = driver;
  }

  public boolean isViewing() {
    return driver.findElement(By.tagName("h2")).getText().trim().toLowerCase().startsWith("oj!");
  }

  public ErrorPage navigateTo() {
    driver.navigate().to(PlayContainer.getBaseUrl() + "/ERRORR");
    return this;
  }
}

File: test/se/crisp/signup4/web/pages/ErrorPage.scala
package se.crisp.signup4.web.pages

import org.openqa.selenium.{By, WebDriver}

class ErrorPage(baseUrl: String, driver: WebDriver) extends Page(baseUrl, driver) {

  def isViewing: Boolean = driver.findElement(By.tagName("h2")).getText.trim.toLowerCase.startsWith("oj!")

  def navigateTo() {
    driver.navigate.to(baseUrl + "/DOESNOTEXIST")
  }
}
